### DISCUSSION TITLE
Global velocity assignation

### DISCUSSION THREAD

**Discussion by Houston4444** [hydrogen-music/hydrogen#1077] [General]
Hi.
I'd like to assign a midi CC event to the global velocity of hydrogen, is it possible ?
I see the velocity automation line, so I guess that if it doesn't exists yet, it would be quite easy to implement. But maybe it already exists and I miss something.

Cheers.

**Comment by theGreatWhiteShark** [comment]
There is a `MASTER_VOLUME_ABSOLUTE` and `MASTER_VOLUME_RELATIVE` `MidiAction` in the code. This might be what you are looking for.

**Comment by Houston4444** [comment]
Thanks.
No, this is not this, I don't wan't to affect the master volume but the general velocity. This can be very different if an instrument plays different samples when it is played with different velocities.

However, I just tried  MASTER_VOLUME_RELATIVE but it seems to not works correctly.
Master volume is down at ~100 /127 , worse, it decreases with CC value but never increase. I probably should open an issue.

**Comment by theGreatWhiteShark** [comment]
Haha. I wasn't aware the automation path does affect the velocity of each individual note. But you are right. Well, there's even states "velocity" to the left of the AP... Anyway.

I can't recall that there is something like a general velocity handled in Hydrogen (apart from the Automation Path). The `AutomationPath` itself needs a larger overhaul anyway (I try to open an issue for this). Maybe we can introduce MIDI/OSC commands altering is as well. But I'm a little bit skeptical here. Because of introducing a new point on the graph will also affect all notes up to the last point. We will see.

 > However, I just tried MASTER_VOLUME_RELATIVE but it seems to not works correctly.
Master volume is down at ~100 /127 , worse, it decreases with CC value but never increase. I probably should open an issue.

Yes, please
